frontend/public/components/utils/kebab.tsx

```diff
@@ -218,42 +218,73 @@ export const ResourceKebab = connectToModel((props: ResourceKebabProps) => {
   />;
 });
 
-export class Kebab extends DropdownMixin {
-  static factory: KebabFactory = kebabFactory;
-
-  onClick = (event, option: KebabOption) => {
-    event.preventDefault();
-
-    if (option.callback) {
-      option.callback();
-    }
+const KebabHook: React.FC<KebabProps> = ({ options, isDisabled, position, id, children }) => {
+  const [isOpen, setIsOpen] = React.useState(false);
 
-    if (option.href) {
-      history.push(option.href);
-    }
+  function onToggle(open) {
+    setIsOpen(open);
+  }
 
-    this.hide();
-  };
+  function onSelect() {
+    isOpen ? setIsOpen(false) : setIsOpen(true);
+  }
 
-  onHover = () => {
+  function onHover() {
     // Check access when hovering over a kebab to minimize flicker when opened.
     // This depends on `checkAccess` being memoized.
-    _.each(this.props.options, (option: KebabOption) => {
+    _.each(options, (option: KebabOption) => {
       if (option.accessReview) {
         checkAccess(option.accessReview);
       }
     });
   }
 
-  render() {
-    const {options, isDisabled} = this.props;
+  let items = [];
+  if (options && options.length){
+    options.forEach((option) => {
+      items.push(
+        <DropdownItem key={option.label}
+          isDisabled={isDisabled}
+          onClick={e => {
+            e.preventDefault();
+            if (option.callback){
+              option.callback();
+            }
+            if (option.href) {
+              history.push(option.href);
+            }
+          }}
+          href={option.href}
+          component={option.href ? 'a' : 'button'}>
+          {option.label}
+        </DropdownItem>
+      );
+    });
+  } else {
+    //default to children if no options are passed, allowing custom jsx for items
+    items = children;
+  }
+
+  return (
+    <Dropdown
```

Comment:
We had a single implementation for kebabs and actions menus that this change undoes.

Reply:
will open a separate PR and try to resolve this piece there for now

Reply:
demo of the new kebab styled w/ pf4 table:
![kebab-dropdown-pf4-tables](https://user-images.githubusercontent.com/4237045/58587735-20aea880-822c-11e9-9628-43c5f735f8b3.png)

Reply:
started a PR for this in [#1629](https://github.com/openshift/console/pull/1629). Will resolve there first.

Reply:
this is proving to be a can o' worms. I've split the logic here for now. Will rebase later if we can complete #1629 subsequently.